src/modules/PackageSlideShow.rb

```diff
@@ -188,150 +214,203 @@ def UpdateTotalProgress(_silent_check)
       UpdateTotalProgressValue()
     end
 
-    # Progress display update
-    # This is called via the packager's progress callbacks.
+    # For API backwards compatibility. DEPRECATED.
     #
-    # @param [Fixnum] pkg_percent  package percentage
+    def DisplayGlobalProgress
+      log.warning "DEPRECATED. Use UpdateTotalProgressText() instead."
+      UpdateTotalProgressText()
+    end
+
+    # Update the total progress text (not the value!).
+    #
+    def UpdateTotalProgressText
+      if @active_downloads > 0 && !parallel_download?
+        UpdateDownloadProgressText()
+      else
+        UpdateInstallationProgressText()
+      end
+    end
+
+    # Update the total progress text for downloading.
+    # This should only be used if parallel download + installation is not in effect.
     #
-    def UpdateCurrentPackageProgress(pkg_percent); end
+    def UpdateDownloadProgressText
+      SlideShow.SetGlobalProgressLabel(
+        _("Downloading...") +
+        Builtins.sformat(
+          # TRANSLATORS: This is about a remaining download size.
+          # %1 is the remaining size with a unit (kiB, MiB, GiB etc.),
+          # %2 the total download size, also with a unit.
+          _(" (Remaining: %1 of %2)"),
+          String.FormatSize(@expected_total_download_size - CurrentDownloadSize()),
+          String.FormatSize(@expected_total_download_size)
```

Comment:
As commented in IRC, I am used to counter up. For example, browsers usually have: "Downloading 8/489 MiB". Somehow, the current version with remaining makes me mentally process it. What do you think?